src/main/java/io/leonis/torch/Torch.java

```diff
@@ -0,0 +1,63 @@
+package io.leonis.torch;
+
+import com.googlecode.lanterna.TextColor;
+import com.googlecode.lanterna.gui2.*;
+import com.googlecode.lanterna.screen.Screen;
+import com.googlecode.lanterna.screen.TerminalScreen;
+import com.googlecode.lanterna.screen.VirtualScreen;
+import com.googlecode.lanterna.terminal.DefaultTerminalFactory;
+import com.googlecode.lanterna.terminal.Terminal;
+import io.leonis.torch.component.TextBackground;
+import io.leonis.torch.component.graph.Gradient;
+import io.leonis.torch.component.graph.LineGraph;
+import io.leonis.torch.component.graph.LineType;
+import io.leonis.torch.handlers.CycleWindowHandler;
+import io.leonis.torch.handlers.MoveWindowHandler;
+
+import java.awt.*;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * The Class Torch.
+ *
+ * @author Jeroen de Jong
+ */
+public final class Torch {
+  /**
+   * Constructs a new window manager.
+   */
+  public Torch(final Function<WindowBasedTextGUI, WindowBasedTextGUI> GUI) throws IOException {
```

Comment:
This function might become a Consumer or smth. That way the side effecting isn't masked.

Reply:
The code in the constructor should be moved to a `#run` method as well though.